README


This archive contains the distribution Dist-Zilla-Plugin-Test-Compile,
version 2.051:

  Common tests to check syntax of your modules, only using core modules

This software is copyright (c) 2009 by Jérôme Quelin.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.020.



Changes
Revision history for Dist-Zilla-Plugin-Test-Compile

2.051     2014-10-18 01:51:02Z
 - fix needs_display tests failing on MSWin32 since 2.048

2.050     2014-10-15 23:57:04Z
 - fix test (in our own test suite) that will break with an upcoming
   Test::More release

2.049     2014-10-14 04:46:44Z
 - use documented Test::Builder interface to examine test details

2.048     2014-10-11 21:54:16Z
 - make tests (and generated test) run when needs_display=1 and $DISPLAY is
   not set

2.047     2014-10-10 05:19:20Z
 - avoid use of Test::More::explain in generated test when version is too old

2.046     2014-08-20 15:10:15Z
 - When warnings occur under fail_on_warning, diag them. (GH #9, Kent Fredric)

2.045     2014-08-07 04:25:14Z
 - re-release to remove README.pod from shipped dist

2.044     2014-08-03 01:11:31Z
 - clean up dynamic prerequisite calculations
 - munge the file properly even if [ExtraTests] has already moved it

2.043     2014-06-13 21:00:24Z
 - fix remaining failures on 5.10.0 (due to buggy -C flags) introduced in
   2.041

2.042     2014-06-04 02:28:25Z
 - fix test failures on older perls introduced in 2.041

2.041     2014-05-30 17:54:57Z
 - properly handle scripts with shebang lines with dashes in the path name, or
   extra whitespace, or comments (RT#95897)

2.040     2014-04-06 18:54:08Z
 - fix bad x_breaks metadata
 - line numbers in shipped code are now almost the same (within 3) as the
   repository source, for easier debugging

2.039     2013-12-02 05:23:01Z
 - fix bad handling of $stderr handle on win32 (from v2.038)

2.038     2013-11-29 18:20:37Z
 - test now reuses the same $stdin and $stderr handles across all processes
 - now asserting minimum perl version required (5.6) in the generated test
 - new 'file' option, for adding extra files to be checked

2.037     2013-10-18 01:18:15Z (Karen Etheridge)
 - files are loaded from lib/ if blib/ does not exist
 - now properly falling back to lib/ for scripts too

2.036     2013-10-12 21:32:51Z (Karen Etheridge)
 - new "xt_mode" option (thanks, David Golden!)

2.035     2013-10-12 17:35:49Z (Karen Etheridge)
 - new "phase" option, for selecting how prereqs are injected (thanks, David
   Golden!)

2.034     2013-10-06 06:24:12Z (Karen Etheridge)
 - now checking the full, unshifted value of $? (thanks, ribasushi!)
 - fixed incomplete prerequisite specification on Test::Warnings

2.033     2013-09-20 23:52:40Z (Karen Etheridge)
 - revert to IPC::Open3 implementation from v2.031 and earlier, but with
   waitpid ordering fixed so we never hang (thanks haarg!)
 - now bootstrapping ourselves so our own compile test is generated with the
   version of code we are shipping, rather than version N-1

2.032     2013-09-20 01:02:20Z (Karen Etheridge)
 - add missing prereq injection for File::Spec, when scripts are being tested
 - redo entire implementation, reverting from IPC::Open3 back to a capture{},
   using an inlined form of the guts of Capture::Tiny courtesy of David
   Golden, to address ongoing occasional hang issues on win32 (see RT#88783
   for the longer version)

2.031     2013-09-19 04:49:51Z (Karen Etheridge)
 - bypass cmd.exe on win32 to try to avoid occasional hang issue (RT#88783)

2.030     2013-09-11 05:01:26Z (Karen Etheridge)
 - depend on fixed Test::Warnings, where ':no_end_tests' also covers the
   done_testing case (which turned out to be a red herring)

2.029     2013-09-10 01:19:46Z (Karen Etheridge)
 - fix test failures for non-authors (oops)

2.028     2013-09-10 01:12:53Z (Karen Etheridge)
 - fix binmode issue on win32 (introduced in 2.026)
 - warnings tests bypassed during installation, to prevent installation issues
   in the presence of deprecation warnings from upstream dependencies (in this
   case, via Moose 2.1100)
 - new 'filename' option, for creating the compile test in a different
   location

2.027     2013-09-08 16:36:45Z (Karen Etheridge)
 - promise in documentation that the generated test requires no non-core
   modules, and is compatible with perl 5.6.2
 - re-release with properly encoded META.yml

2.026     2013-09-07 00:47:15Z (Karen Etheridge)
 - more fixes (hopefully the last) for script checking on win32

2.025     2013-09-04 01:32:34Z (Karen Etheridge)
 - more thoroughly test the generated test that no non-core modules are
   required, and syntax does not require a perl later than 5.8.8
 - fix use of regex syntax in script checking that required perl 5.9.5
 - properly screen out expected warnings from perl -c on older perls

2.024     2013-09-03 00:02:21Z (Karen Etheridge)
 - update warning regexp in test yet again, due to win32 failures (does not
   concern the generated test)

2.023     2013-08-30 01:27:54Z (Karen Etheridge)
 - now routing subprocesses' stdout to our stderr rather than throwing it
   away, for greater visibility into any side effects caused by loading
   modules
 - removed use of Test::Script entirely for testing scripts - now instead we
   call perl -c ourselves
 - we can now properly test scripts using special flags such as -T
 - now skipping non-perl scripts, by checking shebang lines
 - now any version of Test::More can be used (if the bail_out_on_fail option
   is not set)

2.022     2013-08-29 02:04:55Z (Karen Etheridge)
 - documentation fixes and cleanup
 - added missing test prereq on Test::Script (but still not injected into the
   built dist unless scripts are being shipped)
 - stop looking for newlines during install-time testing, to avoid needless
   failure on win32 (does not concern the generated test)

2.021     2013-08-27 00:59:49Z (Karen Etheridge)
 - fix incomplete prereq declaration on Test::Warnings
 - work around dubious install-time test failure on win32 with absolute
   filenames in tests (does not concern the generated test)

2.020     2013-08-24 18:06:49Z (Karen Etheridge)
 - fixed test prereq for Module::CoreList (thanks ambs!)
 - removed long-deprecated [CompileTests]

2.019     2013-08-24 04:09:07Z (Karen Etheridge)

t/compile-tests/lib/Bar.pm
#
# This file is part of Dist-Zilla-Plugin-Test-Compile
#
# This software is copyright (c) 2009 by Jerome Quelin.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package Bar;
# ABSTRACT: Bar

warn 'warning during compilation';
1;
__END__


t/test-compile/lib/Baz/Quz.pm
#
# This file is part of Dist-Zilla-Plugin-Test-Compile
#
# This software is copyright (c) 2009 by Jerome Quelin.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package Baz::Quz;
# ABSTRACT: Baz::Quz
1;
__END__


Build.PL
# This Build.PL for Dist-Zilla-Plugin-Test-Compile was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.009
use strict;
use warnings;

if (eval 'use Module::Build::Tiny 0.039; 1')
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-Test-Compile was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.007.
    use strict;
    use warnings;

    use 5.008;
    # use Module::Build::Tiny 0.039;
    Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "Module::Build::Tiny" => "0.039"
      },
      "dist_abstract" => "Common tests to check syntax of your modules, only using core modules",
      "dist_author" => [
        "J\x{e9}r\x{f4}me Quelin"
      ],
      "dist_name" => "Dist-Zilla-Plugin-Test-Compile",
      "dist_version" => "2.051",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::Test::Compile",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Data::Section" => "0.004",
        "Dist::Zilla::File::InMemory" => 0,
        "Dist::Zilla::Plugin::MetaResources" => "4.300039",
        "Dist::Zilla::Role::FileFinderUser" => 0,
        "Dist::Zilla::Role::FileGatherer" => 0,
        "Dist::Zilla::Role::FileMunger" => 0,
        "Dist::Zilla::Role::PrereqSource" => 0,
        "Dist::Zilla::Role::TextTemplate" => 0,
        "Moose" => 0,
        "Moose::Util::TypeConstraints" => 0,
        "Path::Tiny" => 0,
        "Sub::Exporter::ForMethods" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.008",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "CPAN::Meta::Check" => "0.007",
        "CPAN::Meta::Requirements" => 0,
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "File::pushd" => 0,
        "IO::Handle" => 0,
        "IPC::Open3" => 0,
        "Module::CoreList" => "2.77",
        "Perl::PrereqScanner" => "1.016",
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::MinimumVersion" => 0,
        "Test::More" => "0.94",
        "Test::Warnings" => "0.009",
        "if" => 0,
        "utf8" => 0,
        "version" => 0
      }
    );


    my %fallback_build_requires = (
      "CPAN::Meta::Check" => "0.007",
      "CPAN::Meta::Requirements" => 0,
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "File::pushd" => 0,
      "IO::Handle" => 0,
      "IPC::Open3" => 0,
      "Module::CoreList" => "2.77",
      "Perl::PrereqScanner" => "1.016",
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::MinimumVersion" => 0,
      "Test::More" => "0.94",
      "Test::Warnings" => "0.009",
      "if" => 0,
      "utf8" => 0,
      "version" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}


Makefile.PL
#
# This file is part of Dist-Zilla-Plugin-Test-Compile
#
# This software is copyright (c) 2009 by Jérôme Quelin.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
# This Makefile.PL for Dist-Zilla-Plugin-Test-Compile was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.013
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.31.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'ExtUtils::MakeMaker' => '0',
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.008;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Common tests to check syntax of your modules, only using core modules",
  "AUTHOR" => "J\x{e9}r\x{f4}me Quelin",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Test-Compile",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008",
  "NAME" => "Dist::Zilla::Plugin::Test::Compile",
  "PREREQ_PM" => {
    "Data::Section" => "0.004",
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Plugin::MetaResources" => "4.300039",
    "Dist::Zilla::Role::FileFinderUser" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Moose" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "Path::Tiny" => 0,
    "Sub::Exporter::ForMethods" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "CPAN::Meta::Check" => "0.007",
    "CPAN::Meta::Requirements" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::pushd" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Module::CoreList" => "2.77",
    "Perl::PrereqScanner" => "1.016",
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::MinimumVersion" => 0,
    "Test::More" => "0.94",
    "Test::Warnings" => "0.009",
    "if" => 0,
    "utf8" => 0,
    "version" => 0
  },
  "VERSION" => "2.051",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "CPAN::Meta::Check" => "0.007",
  "CPAN::Meta::Requirements" => 0,
  "Data::Section" => "0.004",
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Plugin::MetaResources" => "4.300039",
  "Dist::Zilla::Role::FileFinderUser" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::FileMunger" => 0,
  "Dist::Zilla::Role::PrereqSource" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::pushd" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::CoreList" => "2.77",
  "Moose" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "Path::Tiny" => 0,
  "Perl::PrereqScanner" => "1.016",
  "Sub::Exporter::ForMethods" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::MinimumVersion" => 0,
  "Test::More" => "0.94",
  "Test::Warnings" => "0.009",
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "utf8" => 0,
  "version" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
